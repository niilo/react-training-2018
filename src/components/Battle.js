import React from 'react';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {
  setFirstPlayer,
  setSecondPlayer,
  resetFirstPlayer,
  resetSecondPlayer,
} from '../battle/actions';
import PlayerInput from './PlayerInput';
import PlayerPreview from './PlayerPreview';

const Battle = props => {
  const {match, playerOne, playerTwo} = props;
  return (
    <div>
      <div className="row">
        {!playerOne.name && (
          <PlayerInput
            id="playerOne"
            label="Player One"
            onSubmit={props.setFirstPlayer}
          />
        )}
        {playerOne.image !== null && (
          <PlayerPreview
            avatar={playerOne.image}
            username={playerOne.name}
            onReset={props.resetFirstPlayer}
            id="playerOne"
          />
        )}
        {!playerTwo.name && (
          <PlayerInput
            id="playerTwo"
            label="Player Two"
            onSubmit={props.setSecondPlayer}
          />
        )}
        {playerTwo.image !== null && (
          <PlayerPreview
            avatar={playerTwo.image}
            username={playerTwo.name}
            onReset={props.resetSecondPlayer}
            id="playerTwo"
          />
        )}
      </div>
      {playerOne.image &&
        playerTwo.image && (
          <Link
            className="button"
            to={{
              pathname: `${match.url}/results`,
              search: `?playerOneName=${playerOne.name}&playerTwoName=${
                playerTwo.name
              }`,
            }}
          >
            Battle
          </Link>
        )}
    </div>
  );
};
const mapStateToProps = state => ({
  playerOne: state.battle.playerOne,
  playerTwo: state.battle.playerTwo,
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      setFirstPlayer,
      setSecondPlayer,
      resetFirstPlayer,
      resetSecondPlayer,
    },
    dispatch,
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Battle);
