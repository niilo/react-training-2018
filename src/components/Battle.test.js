import React from 'react';
import renderer from 'react-test-renderer';
import Battle from './Battle';

it('renders default correctly', () => {
  const tree = renderer.create(<Battle />).toJSON();
  expect(tree).toMatchSnapshot();
});
