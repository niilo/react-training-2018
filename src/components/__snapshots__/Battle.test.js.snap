// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default correctly 1`] = `
<div>
  <div
    className="row"
  >
    <form
      className="column"
      onSubmit={[Function]}
    >
      <label
        className="header"
        htmlFor="username"
      >
        Player One
      </label>
      <input
        autoComplete="off"
        id="username"
        onChange={[Function]}
        placeholder="github username"
        type="text"
        value=""
      />
      <button
        className="button"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </form>
    <form
      className="column"
      onSubmit={[Function]}
    >
      <label
        className="header"
        htmlFor="username"
      >
        Player Two
      </label>
      <input
        autoComplete="off"
        id="username"
        onChange={[Function]}
        placeholder="github username"
        type="text"
        value=""
      />
      <button
        className="button"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
